lib/senders/client_name.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const request = require('request')
+const config = require('../../config').getProperties()
+const log = require('./log')('db')
+
+const OAUTH_SERVER = config.oauth.url
+const clientCache = {}
+const FIREFOX_CLIENT = {
+  name: 'Firefox'
+}
+
+/**
+ * Fetches OAuth client data from the OAuth server.
+ * Stores the data into server memory.
+ * @param clientId
+ * @returns {Promise<any>}
+ */
+function fetchClientData(clientId) {
+  log.trace({ op: 'fetchClientData.start' })
+
+  const options = {
+    url: `${OAUTH_SERVER}/v1/client/${clientId}`,
```

Comment:
Just noting to myself that the validator for `clientId` in incoming requests is `isA.string().max(16).regex(/^[a-zA-Z0-9\-]*$/g)`, so this should be safe; if we ever forget to properly validate the incoming service=blah parameter, this could get interestingly malformed.